tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLLiteralTest.java

```diff
@@ -189,8 +190,18 @@ public void testXMLNegativeSemantics() {
         BAssertUtil.validateError(negativeResult, index++, "incompatible types: expected " +
                 "'xml:Text', found 'xml<xml>'", 124, 19);
         BAssertUtil.validateError(negativeResult, index++, "missing xml CDATA end token", 128, 49);
-
         BAssertUtil.validateError(negativeResult, index++, "xml namespaces cannot be interpolated", 133, 45);
+        BAssertUtil.validateError(negativeResult, index++,
+                "incompatible types: expected 'xml', found 'string'", 140, 60);
+        BAssertUtil.validateError(negativeResult, index++,
+                "incompatible types: expected 'xml', found 'string'", 141, 60);
+        BAssertUtil.validateError(negativeResult, index++,
+                "incompatible types: expected 'xml', found 'int'", 142, 69);
+        BAssertUtil.validateError(negativeResult, index++,
+                "invalid literal for type 'xml': raw templates can only be assigned to " +
+                        "abstract subtypes of 'ballerina/lang.object:0.0.0:RawTemplate'", 143, 52);
+        BAssertUtil.validateError(negativeResult, index++,
```

Comment:
Why does it say RawTemplate[] instead of XML?

Reply:
I thought we are making CET `xml` for query-expr only (https://github.com/ballerina-platform/ballerina-lang/pull/36287/files#diff-f4fbb6e8ffab357b74e188781ad3f6e3235f8e48b64fba8dfb6ad72bf9ccaf93R7566)

Here we have a `type-cast-expr`. Should we do that for all expressions? 
if so, something like the below will not work.

```bal
    int n = 3;
    xml x = xml `<doc>${n}</doc>`;
    io:println(x);
```

Reply:
Yeah, that's true. My concern is for query context, shouldn't the query expression result type be resolved to XML when it's interpolated in an XML context and it should be the type in the error message?